R/pkg_ref_cache_vignettes.R

```diff
@@ -0,0 +1,72 @@
+#' Cache a list of Vignettes files from a package reference
+#'
+#' @inheritParams pkg_ref_cache
+#' @family package reference cache
+#'
+pkg_ref_cache.vignettes <- function(x, name, ...) {
+  UseMethod("pkg_ref_cache.vignettes")
+}
+
+
+
+pkg_ref_cache.vignettes.pkg_remote <- function(x, name, ...) {
+  vignettes_from_url(x$web_url)
+}
+
+
+pkg_ref_cache.vignettes.pkg_install <- function(x, name, ...) {
+  vignettes_from_dir(system.file(package = x$name))
+}
+
+
+
+pkg_ref_cache.vignettes.pkg_source <- function(x, name, ...) {
+  vignettes_from_dir(x$path)
+}
+
+
+
+#' Build a list of Vignettes files discovered within a given directory
+#'
+#' @param path a package directory path expected to contain Vignettes files
+#'
+#' @return a list of parsed Vignettes files
+#'
+vignettes_from_dir <- function(path) {
+  # accommodate unique vignettes files
+
+  folder <- c(source = "/vignettes", boundle = "/inst/doc", binary = "/doc")
+  files <- unlist(lapply(paste0(path, folder), list.files, full.names = TRUE))
+
+  if (!length(files)) return(data.frame())
+
+  file_path = unique(tools::file_path_sans_ext(files))
+  filename = basename(file_path)
+
+  res <- data.frame(filename = filename, path = file_path)
+  res <- res[tolower(res$filename) != tolower("index"), ]
```

Comment:
Instead of returning a `data.frame` which has rows of `filename` and `path`, perhaps this could be just as well represented as a named character vector where the values are the path and the names are the filenames?

I generally prefer to work with as minimal a data structure as possible which really helps to reduce complications if we end up needing to reference this in multiple places.

Reply:
agreed and updated